Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -0,0 +1,741 @@
+/*global define*/
+define([
+        '../Core/Check',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/freezeObject',
+        '../Core/Intersect',
+        '../Core/ManagedArray',
+        '../Core/Math',
+        './Cesium3DTileChildrenVisibility',
+        './Cesium3DTileRefine',
+        './CullingVolume',
+        './OrthographicFrustum',
+        './SceneMode'
+    ], function(
+        Check,
+        defined,
+        defineProperties,
+        freezeObject,
+        Intersect,
+        ManagedArray,
+        CesiumMath,
+        Cesium3DTileChildrenVisibility,
+        Cesium3DTileRefine,
+        CullingVolume,
+        OrthographicFrustum,
+        SceneMode) {
+    'use strict';
+
+    var Cesium3DTilesetTraversal = {};
+
+    function selectTiles(tileset, frameState, outOfCore) {
+        if (tileset.debugFreezeFrame) {
+            return;
+        }
+
+        var maximumScreenSpaceError = tileset._maximumScreenSpaceError;
+
+        tileset._desiredTiles.length = 0;
+        tileset._selectedTiles.length = 0;
+        tileset._selectedTilesToStyle.length = 0;
+        tileset._hasMixedContent = false;
+
+        // Move sentinel node to the tail so, at the start of the frame, all tiles
+        // may be potentially replaced.  Tiles are moved to the right of the sentinel
+        // when they are selected so they will not be replaced.
+        var replacementList = tileset._replacementList;
+        replacementList.splice(replacementList.tail, tileset._replacementSentinel);
+
+        var root = tileset._root;
+        root.updateTransform(tileset._modelMatrix);
+
+        if (!root.insideViewerRequestVolume(frameState)) {
+            return;
+        }
+
+        root.distanceToCamera = root.distanceToTile(frameState);
+
+        if (getScreenSpaceError(tileset, tileset._geometricError, root, frameState) <= maximumScreenSpaceError) {
+            // The SSE of not rendering the tree is small enough that the tree does not need to be rendered
+            return;
+        }
+
+        root.visibilityPlaneMask = root.visibility(frameState, CullingVolume.MASK_INDETERMINATE);
+        if (root.visibilityPlaneMask === CullingVolume.MASK_OUTSIDE) {
+            return;
+        }
+
+        if (root.contentUnloaded) {
+            root._requestHeap.insert(root);
+            return;
+        }
+
+        if (!tileset.skipLODs) {
+            // just execute base traversal and add tiles to _desiredTiles
+            tileset._baseTraversal.execute(tileset, root, maximumScreenSpaceError, frameState, outOfCore);
+            var leaves = tileset._baseTraversal.leaves;
+            var length = leaves.length;
+            for (var i = 0; i < length; ++i) {
+                tileset._desiredTiles.push(leaves.get(i));
+            }
```

Comment:
When `skipLODs` is false, should the `bivariateVisibilityTest` ever be on? I am seeing some cases where it is.